### ISSUE TITLE
wordpress-mysql-replication: Deployment template validation failed

### ISSUE THREAD

**Issue by hubber30** [Azure/azure-quickstart-templates#2051]
When I try to deploy the template I get the next error:
`Deployment template validation failed: 'The template parameters 'storageAccountNamePrefix' in the parameters file are not valid; they are not present in the original template and can therefore not be provided at deployment time.` @marcvaneijk

**Comment by marcvaneijk** [comment]
@hubber30 Good catch!
@singhkay this issue is present because a nested template resides in a different folder. I looked at the options to solve this, but the only proper way is to have all artifacts related to the deployment reside in the same template deployment folder, which is inline with the best practices. I can copy the artifacts and clean up the template. Agree?

**Comment by hubber30** [comment]
@marcvaneijk 
I was able to deploy the template through portal after editing template and deleting following lines:
` "storageAccountNamePrefix": {
            "type": "string",
            "metadata": {
                "description": "Storage account name prefix for the cluster"
            }`
```
                    },
                    "storageAccountNamePrefix": {
                        "value": "[parameters('storageAccountNamePrefix')]"
```
Getting a couple of errors after deployment as well:
https://www.dropbox.com/s/ehnnmf57t4750vr/a1.jpg?dl=1
https://www.dropbox.com/s/3yk66vjoy8dgu35/a2.jpg?dl=1
https://www.dropbox.com/s/afdm78zq6f64by9/a3.jpg?dl=1
https://www.dropbox.com/s/k8gans7nbtgyoui/a4.jpg?dl=1
https://www.dropbox.com/s/k8gans7nbtgyoui/a4.jpg?dl=1

**Comment by hubber30** [comment]
I think there is also something wrong with MySQL server replication. When I shutdown VM1 my website just stops working. Isn't the point of this template to provide fail-over work? Also when I try to check health of second VM by wget http://10.0.1.5:9200, as identified in read me, it reports that second VM is down.

**Comment by GeoffreyEmery** [comment]
mysql replication is failing. You should take this down immediately until you figure this out.  I just wasted almost a 45 minutes going through @hubber30

**Comment by hubber30** [comment]
@GeoffreyEmery 
Are you saying that this template is defective? Did your set up fail? 
Or did you go through my screenshots and recommend me to stop using based on an error you saw?

**Comment by phi16180** [comment]
Hey Guys! @GeoffreyEmery @marcvaneijk Should we still use this template? Any definitive fix for the storageaccountnameprefix issue?

**Comment by GeoffreyEmery** [comment]
@hubber30  Yes the setup is completely failing but fixing it as you did I validated the cluster is failing over properly. I am not a mysql guy so i just moved over to the the https://github.com/Azure/azure-quickstart-templates/tree/master/mysql-ha-pxc which is percona a mysql fork. Note being said here that this template is of great importance and should be fixed. ClearDB is a horrible alternative .

**Comment by phi16180** [comment]
@GeoffreyEmery Can you please guide me on how to deploy wordpress with the percona template?

**Comment by GeoffreyEmery** [comment]
just launch the percona template that will create a percona vm with cluster of percona db's. Then launch wordpress wherever can be anything from a docker instance to a standard web app https://azure.microsoft.com/en-us/documentation/articles/app-service-web-create-web-app-from-marketplace/
dont worry about the db for now you can delete it after.. 
After it launches just edit the config to point at the percona cluster as if its a mysql db.

**Comment by singhkay** [comment]
@phi16180 Just deployed the wordpress-mysql-replication template and it deploys fine for me

![image](https://cloud.githubusercontent.com/assets/6164178/15875705/3b8c4ec6-2cbe-11e6-8033-b11d2b9231e9.png)

Adding @liupeirong who might know why it's not failing over.

**Comment by GeoffreyEmery** [comment]
@singhkay can you provide the parameters  you used to deploy with?

**Comment by singhkay** [comment]
@GeoffreyEmery deleted the RG but deployed through Portal and left the default parameters alone. For dns name make sure it's unique.

**Comment by phi16180** [comment]
@singhkay @GeoffreyEmery Tried several times but deployment always failed.
When deployed directly from the template available in Azure, it didn't pass the validation. **'The template parameters 'location, storageAccountNamePrefix' in the parameters file are not valid; they are not present in the original template and can therefore not be provided at deployment time.** 
![failed_deployment](https://cloud.githubusercontent.com/assets/17506350/15931817/4159cada-2e27-11e6-959b-379bd1f4f038.PNG)
![failed_deployment_4](https://cloud.githubusercontent.com/assets/17506350/15931848/5a7d5752-2e27-11e6-9fad-ca7deabbc02f.PNG)
Also, it's worth noting that when deploying the template directly from the platform the option to use a new or existing RG is not available.
![rg_options_unavailable](https://cloud.githubusercontent.com/assets/17506350/15931900/8ea571e0-2e27-11e6-8362-972654029e30.PNG)
After deleting the failed deployment and the RG created, I tried with a custom deployment where I removed the location and storageAccountNamePrefix parameters in the template. It also failed with the error message: "**The subscription is not registered for the resource type 'components' in the location 'eastus'**."
![failed_custom_deployment_website](https://cloud.githubusercontent.com/assets/17506350/15935124/3ae59550-2e34-11e6-93c4-1f0abfbea2f3.PNG)

As you got it right maybe you can tell me how you proceeded.Thanks.

**Comment by singhkay** [comment]
@phi16180 Looks like you're deploying from somewhere other than https://github.com/Azure/azure-quickstart-templates/tree/master/wordpress-mysql-replication. Is that correct?

**Comment by phi16180** [comment]
Yes.I'm deploying directly from Azure portal.

**Comment by phi16180** [comment]
@singhkay I'm deploying directly from Azure Portal.

**Comment by GeoffreyEmery** [comment]
totally agree i again wasted 30 minutes trying to get this to work to yet another failure.. do you guys validate these?

##error 1
`At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details. (Code: DeploymentFailed)
Deployment template validation failed: 'The template parameters 'location, storageAccountNamePrefix' in the parameters file are not valid; they are not present in the original template and can therefore not be provided at deployment time. The only supported parameters for this template are 'dnsName, vmUserName, vmPassword, mysqlRootPassword, mysqlReplicationPassword, mysqlProbePassword, vmSize, storageAccountType, virtualNetworkName, vnetNewOrExisting, virtualNetworkExistingRGName, dbSubnetName, vnetAddressPrefix, dbSubnetAddressPrefix, dbSubnetStartAddress, imagePublisher, imageOffer, imageSKU, mysqlFrontEndPort0, mysqlFrontEndPort1, sshNatRuleFrontEndPort0, sshNatRuleFrontEndPort1, mysqlProbePort0, mysqlProbePort1, artifactsPath, customScriptCommandToExecute, publicIPName'. Please see https://aka.ms/arm-deploy/#parameter-file for usage details.'. (Code: InvalidTemplate)`

![image](https://cloud.githubusercontent.com/assets/3301804/15939185/842f7374-2e2b-11e6-9800-9525177ca37d.png)

Location of template in Azure 
![image](https://cloud.githubusercontent.com/assets/3301804/15939304/e87ec5f0-2e2b-11e6-97bc-1d5016f024a5.png)

**Comment by GeoffreyEmery** [comment]
also it would much better if you made a separate team for these templates and deployed them as separate repos. So the issues could be associated to a single template

**Comment by singhkay** [comment]
@GeoffreyEmery @phi16180 I see the issue now. That template in portal is not the one here in Github repo. It is a marketplace template, those are not hosted here in Github. It is likely taking a dependency on the MySQL replication template in Github which needs to be fixed as things in Github change much more quickly than Github. Feel free to submit a support request from the Portal re: that template. I'll try to see if I can find the marketplace template author on my end.

**Comment by phi16180** [comment]
@singhkay @marcvaneijk @GeoffreyEmery When deployed from the template available in GitHub it also fails. StorageAccountNamePrefix. 

![deployment_from_github_template](https://cloud.githubusercontent.com/assets/17506350/15950362/29f68554-2e7c-11e6-823a-90e62b17484c.PNG)

**Comment by singhkay** [comment]
@phi16180 @GeoffreyEmery PR to fix is underway https://github.com/Azure/azure-quickstart-templates/pull/2083

**Comment by marcvaneijk** [comment]
@phi16180 @GeoffreyEmery Fix #2083 is merged. The CI build is successful. Are you able to verify if this solves your issue?

**Comment by marcvaneijk** [comment]
Have received multiple confirmations this template works after merging fix #2083. Closing issue
